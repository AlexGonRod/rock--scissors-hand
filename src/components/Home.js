import React from 'react';
import { Link } from 'react-router-dom'

function Home({setUser, setIsSubmitted}) {

	const handleSubmit = () => {
		
		var { uname } = document.forms[0]

		setIsSubmitted(true)
		setUser(uname.value)
	}

	const renderForm = (
		<div className="form">
			<form>
				<div className="input-container">
					<input type="text" name="uname" required />
				</div>
				<div className="button-container">
					<Link to="/game" className="submit" onClick={() => handleSubmit()}>
						JOIN
					</Link>
				</div>
			</form>
		</div>
	)

	return (
		<div className="login-form">
			<div className="title">Create new player</div>
			 {renderForm}
		</div>
	)
}

export default Home
