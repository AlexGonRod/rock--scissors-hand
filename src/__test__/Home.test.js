import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Home from '../components/Home';

describe('<Home />', () => {
    let wrapper = shallow(<Home />);
    beforeEach(() => {
        wrapper = shallow(<Home />);
    });
    beforeEach(() => {
        wrapper = shallow(<Home />);
    });
    test('<Home /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
});