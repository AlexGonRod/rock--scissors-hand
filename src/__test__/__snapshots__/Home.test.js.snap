// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> <Home /> renders 1`] = `
<div
  className="login-form"
>
  <div
    className="title"
  >
    Create new player
  </div>
  <div
    className="form"
  >
    <form>
      <div
        className="input-container"
      >
        <input
          name="uname"
          required={true}
          type="text"
        />
      </div>
      <div
        className="button-container"
      >
        <Link
          className="submit"
          onClick={[Function]}
          to="/game"
        >
          JOIN
        </Link>
      </div>
    </form>
  </div>
</div>
`;
